tests/QARegressionTests/test_core/test_list_dataset_metadata.py

```diff
@@ -0,0 +1,51 @@
+import os
+import unittest
+from click.testing import CliRunner
+
+from core import list_dataset_metadata
+
+
+class TestCLI(unittest.TestCase):
+    def setUp(self):
+        self.runner = CliRunner()
+
+    def test_list_dataset_metadata_with_valid_paths(self):
+        result = self.runner.invoke(
+            list_dataset_metadata, ["-dp", r"tests/resources/test_dataset.xpt"]

```

Comment:
For building file paths we prefer to use os.path.join so the tests work on both windows and linux environments